actionpack/lib/action_controller.rb

```diff
@@ -54,6 +54,12 @@ module ActionController
   autoload :TemplateAssertions, "action_controller/test_case"
 end
 
+ActiveSupport.on_load(:after_initialize) do
+  if config.eager_load
+    ActionController::Base.descendants.each(&:action_methods)
```

Comment:
Instead of iterating over descendants here we could add `ActiveSupport.on_load(:after_initialize)` callback every time new controller is inherited. However that would result to a _ton_ on AS callbacks in case of an large app with hundreds of controllers.

Let me know if you think we'd prefer creating a callback per controller.